beacon-chain/blockchain/process_attestation_helpers.go

```diff
@@ -13,15 +13,22 @@ import (
 	stateTrie "github.com/prysmaticlabs/prysm/beacon-chain/state"
 	"github.com/prysmaticlabs/prysm/shared/attestationutil"
 	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/mputil"
 	"github.com/prysmaticlabs/prysm/shared/params"
 )
 
 // getAttPreState retrieves the att pre state by either from the cache or the DB.
 func (s *Service) getAttPreState(ctx context.Context, c *ethpb.Checkpoint) (*stateTrie.BeaconState, error) {
+	// Use a multilock to allow scoped holding of a mutex by a checkpoint root + epoch
+	// allowing us to behave smarter in terms of how this function is used concurrently.
+	lock := mputil.NewMultilock(string(c.Root) + fmt.Sprintf("%d", c.Epoch))
```

Comment:
```suggestion
	lock := mputil.NewMultilock(fmt.Sprintf("%s%d", c.Root, c.Epoch))
```
Any reason to use concatenation?